CMakeLists.txt

```diff
@@ -82,7 +82,7 @@ option(DISABLE_RCT2 "WIP: Try building without using code and data segments from
 option(USE_MMAP "Use mmap to try loading rct2's data segment into memory.")
 option(LAUNCHPAD_BUILD "Sets some build system options on launchpad.net")
 
-set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations")
+set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Warray-bounds -Waddress -Wchar-subscripts -Wenum-compare")
```

Comment:
@janisozaur: The addition of `-Warray-bounds` in combination with `-Werror` breaks compilation on my Arch Linux system with gcc 6.1.1.

```
[  2%] Building C object CMakeFiles/openrct2.dir/src/windows/editor_inventions_list.c.o
/somewhere/OpenRCT2/src/windows/editor_inventions_list.c: In function 'window_editor_inventions_list_open':
/somewhere/OpenRCT2/src/windows/editor_inventions_list.c:423:53: error: array subscript is below array bounds [-Werror=array-bounds]
  if (_editorInventionsListDraggedItem == beforeItem - 1)
                                          ~~~~~~~~~~~^~~
cc1: all warnings being treated as errors
```

Btw.: The line in question is from 2014.